packages/expo-updates/android/src/main/java/expo/modules/updates/statemachine/UpdatesStateContext.kt

```diff
@@ -0,0 +1,45 @@
+package expo.modules.updates.statemachine
+
+import org.json.JSONObject
+
+/**
+The state machine context, with information intended to be consumed by application JS code.
+ */
+data class UpdatesStateContext(
+  var isUpdateAvailable: Boolean = false,
+  var isUpdatePending: Boolean = false,
+  var isRollback: Boolean = false,
+  var isChecking: Boolean = false,
+  var isDownloading: Boolean = false,
+  var isRestarting: Boolean = false,
+  var latestManifest: JSONObject? = null,
+  var downloadedManifest: JSONObject? = null,
+  var checkError: UpdatesStateError? = null,
+  var downloadError: UpdatesStateError? = null
+) {
+
+  val json: Map<String, Any>
+    get() {
+      val map: MutableMap<String, Any> = mutableMapOf(
+        "isUpdateAvailable" to isUpdateAvailable,
+        "isUpdatePending" to isUpdatePending,
+        "isRollback" to isRollback,
+        "isChecking" to isChecking,
+        "isDownloading" to isDownloading,
+        "isRestarting" to isRestarting
+      )
+      if (latestManifest != null) {
+        map["latestManifest"] = latestManifest!!
```

Comment:
By using `val` in the fields, no need for this non-null-assertion since the value is immutable. (ditto with others below)

```suggestion
        map["latestManifest"] = latestManifest
```